ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -353,24 +365,43 @@ writeSnapshot hasFS encLedger encRef ss cs = do
     encode :: ChainSummary l r -> Encoding
     encode = encodeChainSummary encLedger encRef
 
--- | Delete snapshot from disk
-deleteSnapshot :: HasCallStack => HasFS m h -> DiskSnapshot -> m ()
-deleteSnapshot HasFS{..} = removeFile . snapshotToPath
+-- | Delete snapshot from disk, unless it has a suffix
+deleteSnapshot :: (HasCallStack, Monad m) => HasFS m h -> DiskSnapshot -> m ()
+deleteSnapshot HasFS{..} ds@DiskSnapshot {..} =
+    when (isNothing dsSuffix) $ removeFile $ snapshotToPath ds
 
 -- | List on-disk snapshots, most recent first
 listSnapshots :: Monad m => HasFS m h -> m [DiskSnapshot]
 listSnapshots HasFS{..} =
     aux <$> listDirectory (mkFsPath [])
   where
     aux :: Set String -> [DiskSnapshot]
-    aux = List.sortBy (flip compare) . mapMaybe snapshotFromPath . Set.toList
+    aux = List.sortBy (flip compare) . mapMaybe snapshotFromPath  . Set.toList
```

Comment:
I noticed that this changed a bit which snapshots are deleted. ie if we have
100
100_a
200

previously it would compare based on the Ord instance of `DiskSnapshot`  (not just the slot) and it would delete the 100. now since they have the same slot, it's undeterministic (on my manual tests it deletes nothing).

Reply:
We can change the `Ord` instance to prefer a named snapshot over an unnamed one with the same slot number (and document it!).

For deleting, we should filter out the named ones first as I said elsewhere, so it shouldn't play a role there.